src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,865 @@
+    return {
+      maxReplicas: this.maxReplicas,
+      minReplicas: this.minReplicas,
+      scaleTargetRef: {
+        apiVersion: this.target.apiVersion,
+        kind: this.target.kind,
+        name: this.target.name,
+      },
+      metrics: this.metrics?.map(m => m?._toKube()),
+      behavior: {
+        scaleUp: {
+          policies: this.scaleUp?.policies?.map(p => ({
+            type: p.replicas.type,
+            value: p.replicas.value,
+            periodSeconds: p.duration?.toSeconds() ?? Duration.seconds(15).toSeconds(),
+          })),
+          selectPolicy: this.scaleUp?.strategy,
+          stabilizationWindowSeconds: this.scaleUp?.stabilizationWindow?.toSeconds(),
+        },
+        scaleDown: {
+          policies: this.scaleDown?.policies?.map(p => ({
+            type: p.replicas.type,
+            value: p.replicas.value,
+            periodSeconds: p.duration?.toSeconds() ?? Duration.seconds(15).toSeconds(),
+          })),
+          selectPolicy: this.scaleDown?.strategy,
+          stabilizationWindowSeconds: this.scaleDown?.stabilizationWindow?.toSeconds(),
+        },
+      },
+    };
+  }
+}
+
+
+/**
+ * Metric source types.
+ */
+export enum MetricSourceType {
+  /**
+   * A container metric that will be tracked across all pods of the current scale target.
+   * Only a single container may be tracked in each pod. In other words, a pod
+   * that hosts two of the same container, will be ignored.
+   */
+  CONTAINER_RESOURCE = 'ContainerResource',
+  /**
+   * A global metric that is not associated with any Kubernetes object.
+   * Allows for autoscaling based on information coming from components running outside of
+   * the cluster.
+   */
+  EXTERNAL = 'External',
+  /**
+  * Metric that describes a kubernetes object
+  */
+  OBJECT = 'Object',
+  /**
+   * A pod metric that will be averaged across all pods of the current scale target.
+   */
+  PODS = 'Pods',
+  /**
+   * A resource metric can be used to track the current available resources of a
+   * scale target.
+   */
+  RESOURCE = 'Resource',
+}
+
+/**
+ * Base options for a Metric
+ */
+export interface MetricOptions {
+  /**
+   * The target metric value that will trigger scaling.
+   */
+  readonly target: MetricTarget;
+  /**
+  * The name of the metric to scale on.
+  */
+  readonly name: string;
+  /**
+   * A selector to find a metric by labels.
+   */
+  readonly labelSelector?: kplus.LabelSelector;
+}
+
+
+/**
+ * Options for `Metric.containerResource()`
+ */
+export interface MetricContainerResourceOptions {
+  /**
+   * Container where the metric can be found.
+   */
+  readonly container: Container;
+  /**
+   * Target metric value that will trigger scaling.
+   */
+  readonly target: MetricTarget;
+}
+
+/**
+ * Options for `Metric.object()`
+ */
+export interface MetricObjectOptions extends MetricOptions {
+  /**
+   * Resource where the metric can be found.
+   */
+  readonly object: Resource;
```

Comment:
Always try accepting interfaces instead of concrete classes. Makes it more flexible. 

```suggestion
  readonly object: IResource;
```